Set allow discharge to 0

FLO-2DJJ opened the pull request:
To review/inspect before merging into master:

- [x] Loading the outfalls dialog set 'allow discharge' to 0 or 1 if it is False/True or Null in _user_swmm_nodes_ table.
- [x] Set _swmm_allow_discharge_ in _user_swmm_nodes_ to 0 or 1 when reading SWMM.INP.
- [x] Move message to message bar when saving inlets.
![imagen](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424575/ff4d6b82-51d0-4077-b3f4-fa5deb6c1fcb)
- [x] Remove substring from messages after saving inlets, outfalls, conduits, pumps, orifices, and weirs.
![imagen](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424575/573bfa0c-bd5e-41a7-87f7-694541e79cd4)

FLO-2DKaren commented:
@FLO-2DJJ 
JJ the messages are mostly wrong.  Many of the reference the swmmoutf.dat file.  They don't need to reference any .dat files.  Set up a generic message that doesn't reference a .dat file like the one you get when you save the storage unit dialog.

FLO-2DJJ commented:
The substring
 _Schematize it from the 'Storm Drain Editor' widget before saving into SWMMOUTF.DAT_
removed from messages after saving inlets, outfalls, conduits, pumps, orifices, and weirs.

FLO-2DJJ commented:
I using this pull request to fix a problem commented on by @FLO-2DKaren in a video in issue #1359:
_Video1 7:20 The pump data is not connecting to the Pump Curves when we import an INP._

The problem is that the _import_storm_drain_INP_file_ method is called from an instance of _StormDrainEditorWidget_ which is not the same as the one in the FLO-2D panel. The instance used in the FLO-2D panel is _self.f2d_widget.storm_drain_editor_.

The new call is here (old call commented out):

```
    def import_inp(self):
        """
        Function to export FLO-2D to SWMM's INP file
        """
        # sd_editor = StormDrainEditorWidget(self.iface, self.f2d_plot, self.f2d_table, self.lyrs)
        # sd_editor.import_storm_drain_INP_file("Choose", True)
        self.f2d_widget.storm_drain_editor.import_storm_drain_INP_file("Choose", True)
```

@rpachaly, could you verify this please. An also, maybe, if the above is correct, do something similar in _def export_inp_.

rpachaly commented:
> I using this pull request to fix a problem commented on by @FLO-2DKaren in a video in issue #1359: _Video1 7:20 The pump data is not connecting to the Pump Curves when we import an INP._
> 
> The problem is that the _import_storm_drain_INP_file_ method is called from an instance of _StormDrainEditorWidget_ which is not the same as the one in the FLO-2D panel. The instance used in the FLO-2D panel is _self.f2d_widget.storm_drain_editor_.
> 
> The new call is here (old call commented out):
> 
> ```
>     def import_inp(self):
>         """
>         Function to export FLO-2D to SWMM's INP file
>         """
>         # sd_editor = StormDrainEditorWidget(self.iface, self.f2d_plot, self.f2d_table, self.lyrs)
>         # sd_editor.import_storm_drain_INP_file("Choose", True)
>         self.f2d_widget.storm_drain_editor.import_storm_drain_INP_file("Choose", True)
> ```
> 
> @rpachaly, could you verify this please. An also, maybe, if the above is correct, do something similar in _def export_inp_.

You are right JJ, instantiate the way you coded. Both ways will work but your code follows the flo-2d format.

FLO-2DKaren reviewed (approved):
Testing is working.

FLO-2DKaren commented:
@FLO-2DJJ 
Hi JJ,
Are you going to work on all of the video items here?  If so, I will not merge anything until you are ready.

FLO-2DJJ commented:
> Are you going to work on all of the video items here? If so, I will not merge anything until you are ready.

Yes @FLO-2DKaren . I'm working on the other video items. Wait.

FLO-2DJJ commented:
@FLO-2DKaren,
This pull request addressed the remaining issues in the video as follows:

- **(Video1 7:30) Should we Connect the Type 4 list to the Type 4 table editor when we import an INP.**
Type 4 list is not affected by importing INP. It is fine. All contents and functionality remains as before.

- **(Video 2 7:00) Pull request now imports/exports storage curve description from/to INP.**

- **(Video 1 Noemi) Noemi, Can you look at the video at 11:00 and is this OK?**
This is outside the issues of this pull request.

- **(Video 2 8:15) The Outfall editor seems to be connected to the schematized table. I think it should be connected to the user table.**-
When saving outfalls, both, the user layer (_user_swmm_nodes_ table) and the schematized outfalls layer (_swmmoutf_ table) are updated.

Please, merge the pull request if you are satisfied with that, or tell me if you want any other change.

FLO-2DNoemi commented:
( *Video 1 Noemi) Noemi, Can you look at the video at 11:00 and is this OK?*
This is outside the issues of this pull request.

Let me know if you still need me to look at this issue.
@Juan Jose Rodriguez ***@***.***>
@Karen OBrien ***@***.***>

On Wed, May 29, 2024 at 1:22 PM Juan Jose ***@***.***> wrote:

> @FLO-2DKaren <https://github.com/FLO-2DKaren>,
> This pull request addressed the remaining issues in the video as follows:
>
>    -
>
>    *(Video1 7:30) Should we Connect the Type 4 list to the Type 4 table
>    editor when we import an INP.*
>    Type 4 list is not affected by importing INP. It is fine. All contents
>    and functionality remains as before.
>    -
>
>    *(Video 2 7:00) Pull request now imports/exports storage curve
>    description from/to INP.*
>    -
>
>    *(Video 1 Noemi) Noemi, Can you look at the video at 11:00 and is this
>    OK?*
>    This is outside the issues of this pull request.
>    -
>
>    *(Video 2 8:15) The Outfall editor seems to be connected to the
>    schematized table. I think it should be connected to the user table.*-
>    When saving outfalls, both, the user layer (*user_swmm_nodes* table)
>    and the schematized outfalls layer (*swmmoutf* table) are updated.
>
> Please, merge the pull request if you are satisfied with that, or tell me
> if you want any other change.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/pull/1361#issuecomment-2137913606>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE32O3XHSBI5A67ZULMY45DZEYFFLAVCNFSM6AAAAABINNJUWKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMZXHEYTGNRQGY>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>
